yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,369 @@
-        self.xy_plot = xy_plot
-        self.xy_args = xy_args
-        self.ratio = ratio
-        self.space = space
-
-    def fit(self, X, y, **kwargs):
-        """
-        Sets up the X and y variables for the jointplot
-        and checks to ensure that X and y are of the
-        correct data type
-
-        Fit calls draw
+    # TODO: should we couple more closely with Rank2D?
+    correlation_methods = {
+        'pearson': lambda x, y: pearsonr(x,y)[0],
+        'spearman': lambda x, y: 1.0,
+        'covariance': lambda x, y: 1.0,
+        'kendalltau': lambda x, y: 1.0,
+    }
+
+    def __init__(self, ax=None, columns=None, correlation='pearson', kind="scatter",
+                 hist=True, alpha=0.65, joint_kws=None, hist_kws=None, **kwargs):
+        # Initialize the visualizer
+        super(JointPlot, self).__init__(ax=ax, **kwargs)
+        self._xhax, self._yhax = None, None
+
+        # Set and validate the columns
+        self.columns = columns
+        if self.columns is not None and not isinstance(self.columns, (int, str)):
+            self.columns = tuple(self.columns)
+            if len(self.columns) > 2:
+                raise YellowbrickValueError((
+                    "'{}' contains too many indices or is invalid for joint plot"
+                ).format(columns))
+
+        # Seet and validate the correlation
+        self.correlation = correlation
+        if self.correlation not in self.correlation_methods:
+            raise YellowbrickValueError(
+                "'{}' is an invalid correlation method, use one of {}".format(
+                    self.correlation, ", ".join(self.correlation_methods.keys())
+            ))
 
-        Parameters
-        ----------
+        # Set and validate the kind of plot
+        self.kind = kind
+        if self.kind not in {'scatter', 'hex', 'hexbin'}:
+            raise YellowbrickValueError((
+                "'{}' is invalid joint plot kind, use 'scatter' or 'hex'"
+            ).format(self.kind))
 
-        X : ndarray or DataFrame of shape n x 1
-            A matrix of n instances with 1 feature
+        # Set and validate the histogram if specified
+        self.hist = hist
+        if self.hist not in {True, 'density', 'frequency', None, False}:
+                raise YellowbrickValueError((
+                    "'{}' is an invalid argument for hist, use None, True, "
+                    "False, 'density', or 'frequency'"
+                ).format(hist))
+
+        # If hist is True, test the version availability
+        if self.hist in {True, 'density', 'frequency'}:
+            self._layout()
+
+        # Set the additional visual parameters
+        self.alpha = alpha
+        self.joint_kws = joint_kws
+        self.hist_kws = hist_kws
+
+    @property
+    def xhax(self):
+        if self._xhax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._xhax
```

Comment:
xhax is the axes of the histogram for the X-axis that is on the top of the JointPlot. yhax is the histogram for the Y-axis that is to the right of the JointPlot

Reply:
Maybe add the above to the docstrings of `def xhax(self)` and `def yhax(self)`? E.g. 

```
    @property
    def xhax(self):
    """
    Return the axes of the histogram for the top of the JointPlot (X-axis)
    """
    ...
```

Reply:
I've added the docstrings, but note that this is for developers only because of the `@property` decorator. Using `help(oz.xhax)` will return the docstring for the `matplotlib.Axes` object.